### PR TITLE
docs(graphql): Add deprecated directive

### PR DESCRIPTION
**Pull request by bucanero**
Fix DOC-1

### CODE REVIEW

File: content/graphql/directives/index.md [hunk 1]
```diff
@@ -28,6 +28,12 @@ Reference: [Cascade](/graphql/queries/cascade)
 
 Reference: [Custom directive](/graphql/custom/directive)
 
+### @deprecated
+
+The `@deprecated` directive allows you to mark the schema definition of a field or enum value as deprecated with an optional reason.
```
Comment (aaroncarey) [inline content/graphql/directives/index.md:33]:
```suggestion
The `@deprecated` directive lets you mark the schema definition of a field or `enum` value as deprecated, and also lets you provide an optional reason for the deprecation.
```

File: content/graphql/schema/deprecated.md [hunk 2]
```diff
@@ -5,6 +5,20 @@ weight = 8
     parent = "schema"
 +++
 
-Documentation about `@deprecated` directive coming soon.
+The `@deprecated` directive allows you to tag the schema definition of a field or enum value as deprecated with an optional reason.
 
-<!-- @deprecated directive ... we just pass this through to the generated API, so users can deprecate their fields.  Most GraphQL tools will pick this up and give you a warning if you try and use a deprecated field, so we could show an example of that.-->
+By using the `@deprecated` directive, Dgraph will pass this information through to the generated API, so users can deprecate their fields.
```
Comment (aaroncarey) [inline content/graphql/schema/deprecated.md:10]:
```suggestion
When you use the `@deprecated` directive, Dgraph passes through the deprecation and optional reason to the generated API. So, users are notified and can deprecate their use of the deprecated field or `enum` value.
```

**Review by aaroncarey — APPROVED** [review]
LGTM, also two minor suggested edits

File: content/graphql/schema/deprecated.md (hunk 2 above)
Comment (vmrajas) [inline content/graphql/schema/deprecated.md:10]: Can you remove the statement, "Dgraph will pass this information through to the generated API".

**Review by vmrajas — COMMENTED** [review]
We should also be adding a note at the end of https://dgraph.io/docs/graphql/queries/search-filtering/#get-a-single-object mentioning that,
"get API on interfaces which contain fields with @id directive are being deprecated and will be removed in 21.11. Users are advised to use query API instead" .
For more details, https://github.com/dgraph-io/dgraph/pull/7602 .

File: content/graphql/schema/deprecated.md (hunk 2 above)
Comment (bucanero) [reply content/graphql/schema/deprecated.md:10]: Hello @vmrajas, would this alternative be ok?

```suggestion
When you use the `@deprecated` directive, GraphQL users can deprecate their use of the deprecated field or `enum` value.
```

Comment (vmrajas) [reply content/graphql/schema/deprecated.md:10]: Yeah, this will be ok

Comment (bucanero) [reply content/graphql/schema/deprecated.md]: please check the updated version based on Rajas' feedback

**Review by aaroncarey — APPROVED** [review]

File: content/graphql/schema/deprecated.md (hunk 2 above)
Comment (aaroncarey) [reply content/graphql/schema/deprecated.md]: LGTM

**Comment by aaroncarey** [comment]
> We should also be adding a note at the end of https://dgraph.io/docs/graphql/queries/search-filtering/#get-a-single-object mentioning that,
> "get API on interfaces which contain fields with @id directive are being deprecated and will be removed in 21.11. Users are advised to use query API instead" .
> For more details, [dgraph-io/dgraph#7602](https://github.com/dgraph-io/dgraph/pull/7602) .

Since that is a bit outside of the scope of this PR - patching the gap around coverage of the @deprecated annotation - I filed DOC-228 to track this, with the details that you provided. 

If we end up resolving that in this PR, we can close this. But, it probably makes sense to follow-up with another PR, so we can use DOC-228 to tag this issue for follow-up.

**Pull request merged**